app_dart/lib/src/request_handling/cache_request_handler.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:meta/meta.dart';
+import 'package:neat_cache/neat_cache.dart';
+
+import 'package:cocoon_service/cocoon_service.dart';
+
+import 'body.dart';
+
+/// A [RequestHandler] for serving cached responses.
+/// 
+/// High trafficked endpoints that have responses that do not change
+/// based on request are good for caching. Additionally, saves
+/// reading from Datastore which is expensive both timewise and monetarily.
+/// 
+/// Implementing requires a writer that will keep [responseKey] in the cache updated.
+/// This should be [fallbackHandler], but does not need to be.
+@immutable
+class CachedRequestHandler extends RequestHandler<Body> {
```

Comment:
This should probably be `CachedRequestHandler<T extends Body> extends RequestHandler<T>`

Reply:
Added